src/Model.php

```diff
@@ -1350,24 +1332,7 @@ public function asModel(string $class, array $options = []): self
 
     /**
      * Create new model from the same base class
-     * as $this.
-     *
-     * @return static
-     */
-    public function newInstance(string $class = null, array $options = [])
-    {
-        $model = (self::class)::fromSeed([$class ?? static::class], $options);
-
-        if ($this->persistence) {
-            return $this->persistence->add($model); // @phpstan-ignore-line
```

Comment:
This is the essence of newInstance(). At least for me.
It creates new model object and already links it with same persistence of old object.

Reply:
I am aware, I can not be fixed in the constructor as construtor is not aware of the previous instance/persistence.

The solution is to call `new $class($model->persistence)`, eg. pass the persistence manually when needed.